app/src/org/commcare/activities/components/FormEntryInstanceState.java

```diff
@@ -106,8 +103,7 @@ public void loadFromIntent(Intent intent) {
         if (intent.hasExtra(KEY_FORM_RECORD_DESTINATION)) {
             this.mFormRecordDestination = intent.getStringExtra(KEY_FORM_RECORD_DESTINATION);
         } else {
-            Logger.log(LogTypes.SOFT_ASSERT, "Access outside scoped storage in loading current form.");
-            mFormRecordDestination = ODKStorage.FORM_RECORD_PATH;
+            throw new RuntimeException("Access outside scoped storage in loading current form.");
```

Comment:
the wordings of this doesn't make sense with the code to access outside scope storage deleted. Can we just say more on the lines of "Invalid Intent definition, no extra with name 'KEY_FORM_RECORD_DESTINATION` present in the intent"